### ISSUE TITLE
`no-shadow-route-definition` false positive

### ISSUE THREAD

**Issue by bmish** [ember-cli/eslint-plugin-ember#1134]
Hey @raido, your change hasn't been released yet, but I noticed it is causing a bunch of routes in my application to get incorrectly flagged.

Here's a test case with the issue. Would you be able to fix this? Otherwise I'll need to revert this fix before the next release.

```js
this.route('show', { path: '/:marketingEmailId' });
this.route('edit', { path: '/:marketingEmailId/edit' });
```

_Originally posted by @bmish in https://github.com/ember-cli/eslint-plugin-ember/pull/1132#issuecomment-816254737_

**Comment by raido** [comment]
I'll take a look.

**Comment by raido** [comment]
Bug found, has slipped in due refactoring in previous PR and poor test coverage obviously. I am working on the fix.

**Comment by raido** [comment]
PR opened - https://github.com/ember-cli/eslint-plugin-ember/pull/1135 cc @bmish
